src/System.Security.Cryptography.X509Certificates/tests/X509FilesystemTests.Unix.cs

```diff
@@ -66,6 +68,550 @@ public static void VerifyCrlCache()
+                    using (var certC = new X509Certificate2(TestData.ECDsa256Certificate))
+                    using (var certD = new X509Certificate2(TestData.MicrosoftDotComRootBytes))
+                    {
+                        store.Open(OpenFlags.ReadWrite);
+
+                        store.Add(certA);
+                        store.Add(certB);
+                        Assert.True(Directory.Exists(storeDirectory), "Directory.Exists(storeDirectory)");
+
+                        string newName = store.Name;
+
+                        if (!matchCase)
+                        {
+                            newName = newName.ToUpperInvariant();
+                            Assert.NotEqual(store.Name, newName);
+                        }
+
+                        using (X509Store storeClone = new X509Store(newName, store.Location))
+                        {
+                            storeClone.Open(OpenFlags.ReadWrite);
+
+                            X509Certificate2Collection storeCerts = store.Certificates;
+                            X509Certificate2Collection storeCloneCerts = storeClone.Certificates;
+
+                            Assert.Equal(
+                                storeCerts.OfType<X509Certificate2>(),
+                                storeCloneCerts.OfType<X509Certificate2>());
+
+                            store.Add(certC);
+
+                            // The object was added to store, but should show up in both objects
+                            // after re-reading the Certificates property
+                            storeCerts = store.Certificates;
+                            storeCloneCerts = storeClone.Certificates;
+
+                            Assert.Equal(
+                                storeCerts.OfType<X509Certificate2>(),
+                                storeCloneCerts.OfType<X509Certificate2>());
+
+                            // Now add one to storeClone to prove bidirectionality.
+                            storeClone.Add(certD);
+                            storeCerts = store.Certificates;
+                            storeCloneCerts = storeClone.Certificates;
+
+                            Assert.Equal(
+                                storeCerts.OfType<X509Certificate2>(),
+                                storeCloneCerts.OfType<X509Certificate2>());
+                        }
+                    }
+                });
+        }
+        
+        private static void RunX509StoreTest(Action<X509Store, string> testAction)
+        {
+            string storesDirectory = PersistedFiles.GetUserFeatureDirectory("cryptography", "x509stores");
+            string storeName = "TestStore" + Guid.NewGuid().ToString("N");
+            string storeDirectory = Path.Combine(storesDirectory, storeName.ToLowerInvariant());
+
+            if (Directory.Exists(storeDirectory))
+            {
+                Directory.Delete(storeDirectory, true);
+            }
```

Comment:
Any chance this will cause a problem for other tests running concurrently?  What if someone runs these tests on a machine where they care about other certs that have been installed here?